lib/libmage6.py

```diff
@@ -0,0 +1,155 @@
+        login = urljoin(self.base_url, "accounts/scriptlogin")
+
+        # Sanity checks
+        if not login.startswith("http"):
+            msg = "Bad MAGE_BASE_URL. It should starts with http://... or (bette) https://..."
+            self.logger.fatal(msg)
+            raise LibMageException(msg)
+        if not login.startswith("https://"):
+            self.logger.warn("Using clear http to login. You should use ssl (https) to protect your passsword")
+
+        if self._session is not None:
+            self.logger.warn("A session was already defined for this client. You should logout before login again or not login twice")
+
+        self._session = requests.Session()
+        self.logger.info("Loggin to MAGE with url %s " % login)
+
+        try:
+            request = self._session.post(login, data = { "username": self.username, "password": self.password })
+        except requests.ConnectionError, e:
+            self._session = None
+            msg = "Failed to connect to MAGE: %s" % e
+            self.logger.fatal(msg)
+            self.logger.exception(e)
+            raise LibMageException(msg)
+
+        if request.status_code != 200:
+            self._session = None
+            msg = "Cannot log in Mage. Response was %s: %s" % (request.status_code, request.reason)
+            self.logger.fatal(msg)
+            raise LibMageException(msg)
+
+
+    def logout(self):
+        """Be a good citizen, close session"""
+        #TODO: call mage logout URL
+        pass
+
+
+    def run_query(self, query, unique=False):
+        """Execute a query on mage. Return results as json like dict
+        @:param unique: if True, assume that query return only one. It raises if stricly less or more that 1 response it returned. Default is False
+        @:return list of responses. If uniq=True, returns only one response object instead of a list of response.. """
+        base_query = "ref/mql/json/"
+        url = urljoin(self.base_url, base_query)
+        url = urljoin(url, query)
+        if self._session is None:
+            msg = "You must login before trying to run a query"
+            self.logger.fatal(msg)
+            raise LibMageException(msg)
+        self.logger.debug("Running mage query: %s" % url)
+        request = self._session.get(url)
+        if request.status_code != 200:
+            msg = "Cannot query Mage. Response was %s: %s" % (request.status_code, request.reason)
+            self.logger.critical(msg)
+            raise LibMageException(msg)
+        self.logger.debug("Query answer: %s" % request.json())
+        if unique:
+            response = request.json()
+            if len(response) == 1:
+                return response[0]
+            else:
+                msg = "Unique query did not return exactly one result. Got %s" % len(response)
+                self.logger.critical(msg)
+                raise LibMageException(msg)
+        else:
+            response = request.json()
+        return response
```

Comment:
Note to myself: add an API that returns the description of items, so as to be able to do a cast (everything will be strings)